pkg/registry/resource/utils.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/kubernetes/pkg/apis/resource"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+// AuthorizedForAdmin checks if the request is authorized to get admin access to devices
+// based on namespace label
+func AuthorizedForAdmin(ctx context.Context, deviceRequests []resource.DeviceRequest, namespaceName string, nsClient v1.NamespaceInterface, oldDeviceRequests []resource.DeviceRequest) field.ErrorList {
+	var allErrs field.ErrorList
+	adminRequested := false
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAAdminAccess) {
+		// No need to validate unless feature gate is enabled
+		return allErrs
+	}
```

Comment:
I think this function should validate based *only* on the presence of the boolean API field. If the feature is disabled, that field *should* be getting dropped, which will make this a no-op anyway.

If *some* path leads to the field still being set, we don't want to let that in without a namespace label check just because the feature gate was disabled